src/RE/P/Projectile.cpp

```diff
@@ -29,10 +35,200 @@ namespace RE
+	{
+		LaunchData ldata(a_shooter, a_origin, a_angles, a_ammo, a_weap);
+		return Launch(a_result, ldata);
+	}
+
+	RE::ProjectileHandle* Projectile::LaunchArrow(RE::ProjectileHandle* a_result, Actor* a_shooter, TESAmmo* a_ammo, TESObjectWEAP* a_weap) noexcept
+	{
+		NiAVObject* fire_node;
+
+		// SE address is 140607AB0. Probably better to find AE address and call directly
+		auto get_aimed_heading = [](Actor* a) -> float {
+			float aimHeadingCurrent, heading = a->GetHeading(false);
+			a->GetGraphVariableFloat(FixedStrings::GetSingleton()->aimHeadingCurrent, aimHeadingCurrent);
+			return heading - aimHeadingCurrent;
+		};
+
+		// SE address is 140607B10. Probably better to find AE address and call directly
+		auto get_aimed_rotation_X = [](Actor* a) -> float {
+			auto strings = FixedStrings::GetSingleton();
+			bool aim_active;
+			a->GetGraphVariableBool(strings->bAimActive, aim_active);
+			if (!aim_active)
+				return a->data.angle.x;
+
+			float aimPitchCurrent;
+			a->GetGraphVariableFloat(FixedStrings::GetSingleton()->aimPitchCurrent, aimPitchCurrent);
+			return -aimPitchCurrent;
+		};
+
+		// SE address is 14067B3B0. Probably better to find AE address and call directly
+		auto get_magicnode = [](AIProcess* proc, const auto& biped) -> NiAVObject* {
+			return (proc->middleHigh && biped.get()) ? biped->root->GetObjectByName(FixedStrings::GetSingleton()->npcRMagicNode) : nullptr;
+		};
+
+		// SE address is 14067B170. Probably better to find AE address and call directly
+		auto get_weaponnode = [](AIProcess* proc, const auto& biped) -> NiAVObject* {
+			if (auto midhigh = proc->middleHigh) {
+				if (biped.get()) {
+					return biped->root->GetObjectByName(FixedStrings::GetSingleton()->weapon);
+				} else {
+					return midhigh->unk148;
+				}
+			} else {
+				return nullptr;
+			}
+		};
+
+		if (auto proc = a_shooter->currentProcess) {
+			const auto& biped = a_shooter->GetBiped2();
+			fire_node = a_weap->IsCrossbow() ? get_magicnode(proc, biped) : get_weaponnode(proc, biped);
+		} else {
+			// SE address is 140235070. Probably better to find AE address and call directly
+			auto TESObjectWEAP__get_node = [](TESObjectWEAP* weap, NiAVObject* node) -> NiAVObject* {
+				NiAVObject* ans;
+				if (weap->weaponData.flags.all(TESObjectWEAP::Data::Flag::kEmbeddedWeapon) && !weap->embeddedNode.empty()) {
+					ans = node->GetObjectByName(weap->embeddedNode);
+				} else {
+					ans = node->GetObjectByName("ProjectileNode");
+				}
+
+				if (!ans)
+					ans = node->GetObjectByName("##ProjectileNode");
+
+				return ans;
+			};
+
+			fire_node = TESObjectWEAP__get_node(a_weap, a_shooter->IsPlayerRef() ? a_shooter->GetCurrent3D() : a_shooter->Get3D2());
+		}
+
+		NiPoint3      origin;
+		ProjectileRot angles;
+
+		if (fire_node) {
+			origin = fire_node->world.translate;
+			a_shooter->Unk_A0(fire_node, &angles.x, &angles.z, &origin);
```

Comment:
@fenix31415 FYI, this is giving us compile errors since Unk_A0 is virtual in tesobjectrefr.cpp.